From 9cdce0f9bd4f9082728bfc9d5c83282487cec5d1 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 01 Nov 2022 17:46:10 +0200
Subject: [PATCH] [SW-0] gaudi3: remove export header usage

Currently, ROM includes arcboot export header for
enum and structure reference which was done to aviod duplication
of enum and structure definition in export header and rom internal
header file. However any change in export header file impacts
alignment of functions in ROM code which will cause issues for using
them in other FW component. This patch removes ROM code dependency
on export header file by adding required enum and structure in internal
header files.

Change-Id: Id7918457a8c56f5eab0cb3984366cd611a6e12da
---

diff --git a/arch/gaudi3/arc/dev_dma.h b/arch/gaudi3/arc/dev_dma.h
index 5ad12f0..2e8fd40 100644
--- a/arch/gaudi3/arc/dev_dma.h
+++ b/arch/gaudi3/arc/dev_dma.h
@@ -5,7 +5,6 @@
 
 #include <secureboot.h>
 #include <dma.h>
-#include <arcboot_export_dev.h>
 
 #define MAX_DMA_SIZE		0xFFF0
 #define MAX_DMA_SIZE_CRC	(MAX_DMA_SIZE + CRC32_SIZE)
@@ -13,3 +12,23 @@
 #define ADDR_SIZE_ALIGNMENT	16	/* 16 bytes */
 #define MIN_DMA_SIZE		16	/* 16 bytes */
 #define MIN_DMA_SIZE_CRC	(MIN_DMA_SIZE + CRC32_SIZE)
+
+enum dma_error {
+	DMA_SUCCESS = 0,
+	DMA_INVALID_CRC_ERROR,
+	DMA_INVALID_DEST_TYPE,
+	DMA_UNALIGNED_IMG_SIZE,
+	DMA_UNALIGNED_DEST_ADDR,
+	DMA_INVALID_IMG_SIZE,
+	DMA_ERROR_MAX
+};
+
+enum dma_dest_type {
+	DMA_DEST_HBW = 0,
+	DMA_DEST_SRAM,
+	DMA_DEST_MAX
+};
+
+enum dma_error dma_copy_image_v2(uint32_t image_size, uint32_t flash_offset,
+                                 uint64_t dest_addr, enum dma_dest_type dest_type,
+			         bool skip_crc_check);
diff --git a/arch/gaudi3/arc/dev_security.h b/arch/gaudi3/arc/dev_security.h
index 2af93ff..2deee0c 100644
--- a/arch/gaudi3/arc/dev_security.h
+++ b/arch/gaudi3/arc/dev_security.h
@@ -7,7 +7,15 @@
 #define _GAUDI3_DEV_SECURITY_H_
 
 #include <sha512.h>
-#include <arcboot_export_dev.h>
+
+struct rom_rr_ctx {
+	uint8_t rr_idx;
+	uint8_t en_val;
+	uint32_t data_min;
+	uint32_t data_min_hi;
+	uint32_t data_max;
+	uint32_t data_max_hi;
+};
 
 #define NUM_KEYS_IN_ROM					5
 #define SECURITY_DWORD_PER_KEY				12
